### PR TITLE
References code cleanup (candidate edition)

### PR DESCRIPTION
**Pull request by elliotcm**
## Context

We no longer use the "old" references flow, so we're removing that path through the code from the candidate part of the site.

## Changes proposed in this pull request

Removes the feature flag, the "old" code branches of those flag conditionals, moves/renames the "new references" stuff to just "references", and updates many tests to work with the new references flow.

## Guidance to review

We're probably going to have to review this as a group.

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database

### CODE REVIEW

File: app/controllers/candidate_interface/prefill_application_form_controller.rb [hunk 1]
```diff
@@ -50,23 +50,20 @@ def destroy_blank_application
     end
 
     def test_application_options
-      test_application_options = {
+      {
         recruitment_cycle_year: RecruitmentCycle.current_year,
-        states: [:unsubmitted_with_completed_references],
```
Comment (carlosmartinez) [inline app/controllers/candidate_interface/prefill_application_form_controller.rb:55]: need to check whether carry-over applications can be `:unsubmitted_with_completed_references`

File: app/lib/test_applications.rb [hunk 2]
```diff
@@ -374,7 +352,18 @@ def accept_offer(choice, incomplete_references)
     fast_forward
     AcceptOffer.new(application_choice: choice).save!
     @application_form.reload
-    reference_state(incomplete_references) if next_recruitment_cycle_application?(@application_form.recruitment_cycle_year)
+
+    if incomplete_references
+      @application_form.application_references.each do |reference|
+        submit_reference!(reference.reload)
+        reference.update(selected: true)
```
Comment (carlosmartinez) [inline app/lib/test_applications.rb:359]: this can go

File: app/mailers/candidate_mailer.rb [hunk 3]
```diff
@@ -285,7 +285,7 @@ def conditions_met(application_choice)
     course_name = "#{@course.name_and_code} at #{@course.provider.name}"
     email_for_candidate(
       application_choice.application_form,
-      subject: I18n.t!("candidate_mailer.conditions_met.#{application_choice.application_form.show_new_reference_flow? ? 'subject_2023' : 'subject_2022'}", course_name:),
+      subject: I18n.t!('candidate_mailer.conditions_met.subject_2023', course_name:),
```
Comment (carlosmartinez) [inline app/mailers/candidate_mailer.rb:288]: `candidate_mailer.conditions_met.subject`

File: app/queries/get_incomplete_reference_applications_ready_to_nudge.rb [hunk 4]
```diff
@@ -26,13 +26,8 @@ def call
...
 
-    scope = if FeatureFlag.active?(:new_references_flow)
-              ApplicationForm.where(recruitment_cycle_year: ApplicationForm::OLD_REFERENCE_FLOW_CYCLE_YEAR)
-            else
-              ApplicationForm.current_cycle
-            end
-
-    scope
+    ApplicationForm
+      .where(recruitment_cycle_year: ApplicationForm::OLD_REFERENCE_FLOW_CYCLE_YEAR)
```
Comment (carlosmartinez) [inline app/queries/get_incomplete_reference_applications_ready_to_nudge.rb:30]: use current cycle

File: app/services/cancel_outstanding_references.rb [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
-  delegate :application_references, :show_new_reference_flow?, to: :application_form
+  delegate :application_references, to: :application_form
 
   def initialize(application_form:)
     @application_form = application_form
   end
 
   def call!
-    return unless show_new_reference_flow?
+    return if application_form.recruitment_cycle_year <= ApplicationForm::OLD_REFERENCE_FLOW_CYCLE_YEAR
```
Comment (carlosmartinez) [inline app/services/cancel_outstanding_references.rb:11]: let's remove guard clause

File: config/locales/candidate_interface/review_application.yml [hunk 6]
```diff
@@ -69,7 +69,7 @@ en:
     references:
       incomplete: Add %{minimum_references} referees to your application
       complete_section: Add your references
```
Comment (carlosmartinez) [inline config/locales/candidate_interface/review_application.yml:71]: double check

File: spec/system/candidate_interface/references/candidate_sends_a_reminder_email_spec.rb [hunk 7]
```diff
@@ -1,69 +0,0 @@
-require 'rails_helper'
```
Comment (carlosmartinez) [inline spec/system/candidate_interface/references/candidate_sends_a_reminder_email_spec.rb:1]: check if this is replicated

Comment (elliotcm) [reply spec/system/candidate_interface/references/candidate_sends_a_reminder_email_spec.rb:1]: This (and the reference history one) are replicated here:
spec/system/candidate_interface/references/candidate_views_their_references_on_the_post_offer_dashboard_spec.rb

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**